### ISSUE TITLE
LAN resolver has weird loop bugs

### ISSUE THREAD

**Issue by jwheare** [RJ/playdar-core#8]
Issues with results being dispatched in an endess loop and then coming back mulitplied several times.

Here's an example of a response with multi-duplicated results, and also the "port" field repeated lots of times.

http://dl.getdropbox.com/u/175535/playdar/jsonp_result.txt

And here are the log outputs for the two machines on the LAN:
http://dl.getdropbox.com/u/175535/playdar/log_lan_dispatcher.txt
http://dl.getdropbox.com/u/175535/playdar/log_lan_receiver.txt

**Comment by RJ** [comment]
the "port" spam bug should be fixed, pull and test please

**Comment by jwheare** [comment]
Port issue is fixed, but endless LAN results loop is still an issue.

Here's the JSON I get back on the 6th poll (also shown: the resolve call for context)
http://dl.getdropbox.com/u/175535/playdar/lanfail2/jsonp.txt
On every poll, it had more duplicate results, that kept coming in from the other end of the LAN until I killed the daemon on that side. It didn't stop when I killed the daemon on my side.

Here are the playdar.logs:

My side (shows queries getting dispatched and results coming in from other side): http://dl.getdropbox.com/u/175535/playdar/lanfail2/log_lan_dispatcher.txt
Other side (shows queries received and results response getting spammed back at me): 
http://dl.getdropbox.com/u/175535/playdar/lanfail2/log_lan_responder.txt

Worth noting that the endlessly duplicated result items all share the same result structure, same sid etc... so they're not being regenerated each time. The sids aren't just a hash on the file right? They're based on generation time too?
